RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -1423,43 +1560,103 @@ class DeepTauId : public deep_tau::DeepTauBase {
-                                          electrons,
-                                          muons,
-                                          pfCands,
-                                          outer_grid,
-                                          tau_funcs,
-                                          false);
+      dynamic_cast<const TauCastType&>(tau), 
+      tau_index, 
+      tau_ref, 
+      pv, 
+      rho, 
+      tau_funcs
+    );
+    checkInputs(*tauBlockTensor_, "input_tau", dnn_inputs_2017_v2::TauBlockInputs::NumberOfInputs);
+    createConvFeatures<CandidateCastType>(
+      dynamic_cast<const TauCastType&>(tau),
+      tau_index,
+      tau_ref,
+      pv,
+      rho,
+      electrons,
+      muons,
+      pfCands,
+      *inner_grid,
+      tau_funcs,
+      true
+    );
+    checkInputs(*eGammaTensor_[true], "input_inner_egamma", dnn_inputs_2017_v2::EgammaBlockInputs::NumberOfInputs, inner_grid);
+    checkInputs(*muonTensor_[true], "input_inner_muon", dnn_inputs_2017_v2::MuonBlockInputs::NumberOfInputs, inner_grid);
+    checkInputs(*hadronsTensor_[true], "input_inner_hadrons", dnn_inputs_2017_v2::HadronBlockInputs::NumberOfInputs, inner_grid);
+    createConvFeatures<CandidateCastType>(
+      dynamic_cast<const TauCastType&>(tau),
+      tau_index,
+      tau_ref,
+      pv,
+      rho,
+      electrons,
+      muons,
+      pfCands,
+      *outer_grid,
+      tau_funcs,
+      false
+    );
+    checkInputs(*eGammaTensor_[false], "input_outer_egamma", dnn_inputs_2017_v2::EgammaBlockInputs::NumberOfInputs, outer_grid);
+    checkInputs(*muonTensor_[false], "input_outer_muon", dnn_inputs_2017_v2::MuonBlockInputs::NumberOfInputs, outer_grid);
+    checkInputs(*hadronsTensor_[false], "input_outer_hadrons", dnn_inputs_2017_v2::HadronBlockInputs::NumberOfInputs, outer_grid);
+
+    if ( save_inputs ) {
+      std::string json_file_name = Form("DeepTauId_%i.json", file_counter);
+      json_file = new std::ofstream(json_file_name.data());
+      is_first_block = true;
+      (*json_file) << "{";
+      saveInputs(*tauBlockTensor_, "input_tau", dnn_inputs_2017_v2::TauBlockInputs::NumberOfInputs);
+      saveInputs(*eGammaTensor_[true], "input_inner_egamma", dnn_inputs_2017_v2::EgammaBlockInputs::NumberOfInputs, inner_grid);
+      saveInputs(*muonTensor_[true], "input_inner_muon", dnn_inputs_2017_v2::MuonBlockInputs::NumberOfInputs, inner_grid);
+      saveInputs(*hadronsTensor_[true], "input_inner_hadrons", dnn_inputs_2017_v2::HadronBlockInputs::NumberOfInputs, inner_grid);
+      saveInputs(*eGammaTensor_[false], "input_outer_egamma", dnn_inputs_2017_v2::EgammaBlockInputs::NumberOfInputs, outer_grid);
+      saveInputs(*muonTensor_[false], "input_outer_muon", dnn_inputs_2017_v2::MuonBlockInputs::NumberOfInputs, outer_grid);
+      saveInputs(*hadronsTensor_[false], "input_outer_hadrons", dnn_inputs_2017_v2::HadronBlockInputs::NumberOfInputs, outer_grid);
+      (*json_file) << "}";
+      delete json_file;
```

Comment:
I have changed the code such that the JSON file is only created in case the save_inputs_ flag is true. I think it is important to only create the JSON file in case the save_inputs_ flag is true, as a new JSON file will be created for each RecoTau object (the creating of the JSON file is meant for small debugging jobs. In a typical crab jobs, there may be several 100k JSON files created!)